Sources/Sentry/SentryNSFileManagerSwizzling.m

```diff
@@ -0,0 +1,84 @@
+#import "SentryNSFileManagerSwizzling.h"
+#import "SentryCrashDefaultMachineContextWrapper.h"
+#import "SentryCrashMachineContextWrapper.h"
+#import "SentryCrashStackEntryMapper.h"
+#import "SentryDependencyContainer.h"
+#import "SentryFileIOTracker.h"
+#import "SentryInAppLogic.h"
+#import "SentryNSProcessInfoWrapper.h"
+#import "SentryOptions+Private.h"
+#import "SentryStacktraceBuilder.h"
+#import "SentrySwizzle.h"
+#import "SentryThreadInspector.h"
+#import <SentryLog.h>
+#import <objc/runtime.h>
+
+@interface SentryNSFileManagerSwizzling ()
+
+@property (nonatomic, strong) SentryFileIOTracker *dataTracker;
+
+@end
+
+@implementation SentryNSFileManagerSwizzling
+
++ (SentryNSFileManagerSwizzling *)shared
+{
+    static SentryNSFileManagerSwizzling *instance = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{ instance = [[self alloc] init]; });
+    return instance;
+}
+
+- (void)startWithOptions:(SentryOptions *)options
+{
+    self.dataTracker = [[SentryFileIOTracker alloc]
+        initWithThreadInspector:[[SentryThreadInspector alloc] initWithOptions:options]
+             processInfoWrapper:[SentryDependencyContainer.sharedInstance processInfoWrapper]];
+    [self.dataTracker enable];
+
+    if (!options.experimental.enableFileManagerSwizzling) {
+        SENTRY_LOG_DEBUG(@"Experimental auto-tracking of FileManager is disabled")
+        return;
+    }
+    [SentryNSFileManagerSwizzling swizzleNSFileManager];
```

Comment:
`h`: We must only swizzle the file manager on iOS 18, macOS 15 and above, as swizzling can always break something.

Reply:
As mentioned in the other PR, there might be additional swizzling coming up in future OS updates. Therefore the availability checks are inside `swizzleNSFileManager` and not in `startWithOptions`

Reply:
Ah I missed that check. Thanks 👍